packages/@aws-cdk/aws-codedeploy/README.md

```diff
@@ -225,6 +225,31 @@ In order to deploy a new version of this function:
 2. Re-deploy the stack (this will trigger a deployment).
 3. Monitor the CodeDeploy deployment as traffic shifts between the versions.
 
+#### Create a custom Deployment Config
+
+CodeDeploy for Lambda comes with built-in configurations for traffic shifting. If you want to specify your own strategy, you can do so with the CustomLambdaDeploymentConfig
+construct, letting you specify precisely how fast a new function version is deployed.
+
+```ts
+const config = new codedeploy.CustomLambdaDeploymentConfig(stack, 'CustomConfig', {
+  type: codedeploy.CustomLambdaDeploymentConfigType.CANARY,
+  interval: '1',
+  percentage: '5',
+});
+const deploymentGroup = new codedeploy.LambdaDeploymentGroup(stack, 'BlueGreenDeployment', {
+  application,
+  alias,
+  deploymentConfig: config,
+});
+```
+
+When using a CustomLambdaDeploymentConfig, make sure you add a dependency between your deployment group and the deployment config, so it is guaranteed to
+exist before the deployment group is created.
+
+```ts
+deploymentGroup.node.addDependency(config)
```

Comment:
I don't like the fact that you hav to do it "manually"... can we special-case this in `LambdaDeploymentGroup` by checking whether the `deploymentConfig` is a `Construct` and, if yes, adding this dependency ourselves?

Reply:
> Thanks for the update @flemjame-at-amazon !
> 
> I just tried it. While the initial resource deployed correctly, after I changed `interval` in code and tried to deploy it again, I got:
> 
> ```
> Failed to delete resource. User: arn:aws:sts::828671620168:assumed-role/CustomLambdaDeploymentCon-AWS679f53fac002430cb0da5-16SO4DQ1B4DWB/CustomLambdaDeploymentCon-AWS679f53fac002430cb0d
> a5-8FPQ6PM2SR31 is not authorized to perform: codedeploy:DeleteDeploymentConfig on resource: arn:aws:codedeploy:us-west-2:828671620168:deploymentconfig:DeployConfigLambdaCanary20Percent
> 1Minutes
> ```
> 
> And it left the old config in my account, without cleaning it up. So, needs some work still 🙂.

That's odd because mine was working. If you use "AwsCustomResourcePolicy.ANY_RESOURCE" instead of "this.deploymentConfigArn" does it still give you the same error?

Reply:
We now add the dependency ourselves